lib/validator/validator_test.go

```diff
@@ -15,31 +15,405 @@
 package validator_test
 
 import (
+	gonet "net"
+
 	. "github.com/tigera/libcalico-go/lib/validator"
 
 	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
+	"github.com/tigera/libcalico-go/lib/api"
 	"github.com/tigera/libcalico-go/lib/backend/model"
+	"github.com/tigera/libcalico-go/lib/net"
+	"github.com/tigera/libcalico-go/lib/numorstring"
+	"github.com/tigera/libcalico-go/lib/scope"
 )
 
-var _ = DescribeTable("Validator",
-	func(input interface{}, valid bool) {
-		if valid {
-			Expect(Validate(input)).To(BeNil(),
-				"expected value to be valid")
-		} else {
-			Expect(Validate(input)).ToNot(BeNil(),
-				"expected value to be invalid")
-		}
-	},
-	// Empty rule is valid, it means "allow all".
-	Entry("empty rule", model.Rule{}, true),
-
-	// Actions.
-	Entry("should accept allow action", model.Rule{Action: "allow"}, true),
-	Entry("should accept deny action", model.Rule{Action: "deny"}, true),
-	Entry("should accept next-tier action", model.Rule{Action: "next-tier"}, true),
-	Entry("should accept log action", model.Rule{Action: "log"}, true),
-	Entry("should reject unknown action", model.Rule{Action: "unknown"}, false),
-	Entry("should reject unknown action", model.Rule{Action: "allowfoo"}, false),
-)
+func init() {
+	var cidr *gonet.IPNet
+	var ipv4_1, ipv4_2, ipv6_1, ipv6_2 net.IP
+	var netv4_1, netv4_2, netv4_3, netv4_4, netv4_5, netv6_1, netv6_2, netv6_3, netv6_4 net.IPNet
+
+	// We need some pointers to ints, so just define as values here.
+	var V0 = 0
+	var V128 = 128
+	var V254 = 254
+	var V255 = 255
+	var V256 = 256
+
+	// Set up some values we use in various tests.
+	_ = ipv4_1.UnmarshalText([]byte("1.2.3.4"))
+	_ = ipv4_2.UnmarshalText([]byte("100.200.0.0"))
+	_ = ipv6_1.UnmarshalText([]byte("aabb:aabb::ffff"))
+	_ = ipv6_2.UnmarshalText([]byte("aabb::abcd"))
+	_, cidr, _ = gonet.ParseCIDR("1.2.3.4/32")
```

Comment:
In my tests, I solved this initialisation problem by writing a local `mustParseCIDR(string)` function that returns the CIDR only and panics on error.